src/main/java/com/pi4j/drivers/input/GameController.java

```diff
@@ -0,0 +1,109 @@
+package com.pi4j.drivers.input;
+
+import com.pi4j.context.Context;
+import com.pi4j.io.ListenableOnOffRead;
+import com.pi4j.io.gpio.digital.DigitalInput;
+
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A simple digital game controller. Note that not all keys are available on all controllers / hats.
+ * For instance, the Sense hat only provides a Joystick with directional inputs and a "center" click.
+ */
+public class GameController implements Closeable {
+
+    /**
+     * Key names based on a typical simple Game controller and some additional KEY_1 .. KEY_3 labels found on the
+     * Waveshare 1.4" and 1.33" display hats.
+     * <p>
+     * While it would be an option to map "special" hat keys to standard game controller keys, which of these
+     * keys are needed will depend on the application -- so we pass this problem up to the application, supporting
+     * this case with getKeyOrFallback().
+     */
+    public enum Key {
+        LEFT,
+        RIGHT,
+        UP,
+        DOWN,
+        CENTER,
+        A,
+        B,
+        X,
+        Y,
+        SELECT,
+        START,
+        KEY_1,
+        KEY_2,
+        KEY_3
+    }
+
+    /**
+     * A builder that provides a straightforward way to create a controller out of digital inputs.
+     */
+    public static class Builder {
+        private final Context pi4j;
+        private final Map<Key, ListenableOnOffRead<?>> keyMap = new HashMap<>();
+
+        public Builder(Context pi4J) {
+            this.pi4j = pi4J;
+        }
+
+        public Builder addDigitalInput(Key key, int pin) {
```

Comment:
do we not need a debounce? No pull-up/down resistance?

Reply:
Let's adjust this in a followup when things are back to normal -- I can look into the sense hat joystick in the meantime, which seems to be file io based -- similar to the corresponding LED matrix